src/bittorrent_helper.cc

```diff
@@ -718,6 +729,16 @@ const unsigned char* getStaticPeerId()
   }
 }
 
+// If PeerAgent is not generated, it is created with default agent
+// aria2/PACKAGE_VERSION
+const std::string& getStaticPeerAgent()
+{
+  if (peerAgent.empty()) {
+    generateStaticPeerAgent(DEFAULT_PEER_AGENT);
+  }
+  return const_cast<const std::string&>(peerAgent);
```

Comment:
I think you don't have to do const_cat<> here.  Just return peerAgent.  Caller will see const std::string & as per function return type.